### ISSUE TITLE
.NET 7 build

### ISSUE THREAD

**Issue by Wolf-64** [flathub/at.vintagestory.VintageStory#53]
Hi there! Since Tyron stated that the .NET 7 build of Vintage Story is seeing significant performance improvements, I was wondering if it would be possible for you to also package a .NET 7 build as flatpak. 

This would mostly be beneficial for SteamDeck players (myself included), as we rely mostly on flatpaks, esp. when it comes to dependencies like the .NET runtime, which isn't easily available on SteamDeck. And for this kind of lower-end hardware, the improved performance would surely be noticeable and improve playability a lot.

**Comment by mard** [comment]
I made a quick and dirty flatpak prototype that supports .NET 7 builds. Check the branch [here](https://github.com/mard/at.vintagestory.VintageStory/commit/282b3c89f8ec18f46a238c7522fae6ad114a216d). You can use `flatpak-builder` to build it.

I tested [1.17.11-rc.3](https://info.vintagestory.at/v117-net7) against dotnet7, in Singleplayer and a random modded multiplayer server - works great!

I was not able to run latest stable 1.18.7 with it, however. It complains about missing `*.runtimeconfig.json` and `*.deps.json` files, but they are present in .NET 7 build 1.17.11-rc.3.

Possible regressions:

* Verify the certificate situation - Mono used `cert-sync` tool to update mono internal certificate storage. We need to check how .NET 7 manages certificates and verify if a similar solution is still needed in .NET 7.
* Source code mods support - Check if we're still capable of running some mods that are interpreted on-the-run, like [roadworks](https://mods.vintagestory.at/roadworks).

**Comment by mard** [comment]
I did some casual testing today on dotnet7. I'm very satisfied with the results. The game performs much better on my ThinkPad T480, loads games noticeably faster, renders chunks faster and performs better overall. I noticed no problems with loading source source code mods too.

Manifest won't longer need `libgdiplus`, because the support for `System.Drawing` was completely removed from non-Windows systems for .NET 7. `cert-sync` hack also won't be needed, as .NET relies on platform's certificate storage now.

I will be continuing the tests, but so far it's looking really good and doesn't rise up any problems for me. Now, going back to original question...

>I was wondering if it would be possible for you to also package a .NET 7 build as Flatpak.

@Wolf-64 Yes, I built it locally, but packaging and distributing it is more complicated. I'm aware that this manifest can't be updated before .NET 7 builds become stable themselves. I wish I could provide some provisional support for experimental .NET 7 Flatpak builds, but I don't have necessary knowledge or resources to host a whole Flatpak repository by myself. On the upside, .net7 stable Vintage Story is just around the corner - so I think the best is to wait a little bit longer.
